src/js/lightbeamVR.js

```diff
@@ -0,0 +1,77 @@
+// eslint-disable-next-line no-unused-vars
+const lightbeamVR = {
+  init() {
+    this.loadLightbeamVR();
+  },
+
+  loadLightbeamVR() {
+    const vrView = document.getElementById('vr-view-button');
+    vrView.addEventListener('click', async () => {
+      await this.getData();
+      await this.runLightbeam();
+    });
+  },
+
+  transformData(data) {
+    const nodes = [];
+    let links = [];
+    for (const website in data) {
+      const site = data[website];
+      data[website].id = site.hostname;
+      data[website].name = site.hostname;
+      if (site.thirdParties) {
+        const thirdPartyLinks = site.thirdParties.map((thirdParty) => {
+          return {
+            source: website,
+            target: thirdParty
+          };
+        });
+        links = links.concat(thirdPartyLinks);
+      }
+      nodes.push(data[website]);
+    }
+
+    return {
+      nodes,
+      links
+    };
+  },
+
+  async getData() {
+    const data = await storeChild.getAll();
+    const transformedData = this.transformData(data);
+    const blob = new Blob([JSON.stringify(transformedData ,' ' , 2)],
+      {type : 'application/json'});
+    const url = window.URL.createObjectURL(blob);
+    const downloading = browser.downloads.download({
```

Comment:
I know it'd likely be much more convoluted, but this seems like a pretty good use case for a Service Worker, using the Cache API.

some basic Service Workers of how to do that:

- https://serviceworke.rs/json-cache_service-worker_doc.html
- https://googlechrome.github.io/samples/service-worker/mock-responses/service-worker.js
- https://googlechrome.github.io/samples/service-worker/post-message/index.html
- https://serviceworke.rs/virtual-server_index_doc.html

you'd just `postMessage` some data to the worker, stringify it, and have it be returned upon `'fetch'` events whenever requesting `/data.json`.

I could be misunderstanding the flow here, but I think that would enable storing, serving, and updating the JSON file just fine.

let me know if I ought to file a separate issue to elaborate on this. don't want to suggest too much additional work here 💻

Reply:
Lets make this a separate issue, we have this downloading elsewhere in Lightbeam too anyway even if we remove the need here. The data will change a lot. Is there likely to be an advantage on caching this if it might change per minute?
We could cache the JSON some other form of storage too perhaps if Service Workers don't work correctly in web extensions (I have never tried and I know there are some bugs).